src/main/scala/sigmastate/serialization/DataJsonEncoder.scala

```diff
@@ -134,6 +171,40 @@ object DataJsonEncoder {
         }
         val coll = Colls.fromArray(collSource.result())(RType.AnyType)
         Evaluation.toDslTuple(coll, t)
+      case SGroupElement =>
+        val str = decodeBytes(json)
+        val r = SigmaSerializer.startReader(str)
+        DataSerializer.deserialize(SGroupElement, r)
+      case SAvlTree =>
+        val str = decodeBytes(json)
+        val r = SigmaSerializer.startReader(str)
+        DataSerializer.deserialize(SAvlTree, r)
+      case SSigmaProp =>
+        val str = decodeBytes(json)
+        val r = SigmaSerializer.startReader(str)
+        DataSerializer.deserialize(SSigmaProp, r)
+      case SBox =>
+        val value = decodeData(json.hcursor.downField(s"value").focus.get, SLong)
+        val tree = ErgoTreeSerializer.DefaultSerializer.deserializeErgoTree(decodeBytes(json.hcursor.downField(s"tree").focus.get))
+        val tokens = decodeData(json.hcursor.downField(s"tokens").focus.get, SCollectionType(STuple(SCollectionType(SByte), SLong))).asInstanceOf[Coll[(Coll[Byte], Long)]].map {
```

Comment:
"tokens" should be created as PairColl. This may be achieved for free if they are encoded and decoded using recursive call to encodeData/decodeData. This is because tokens have `Coll[(TokenId, Long)]` type, which is legitimate type for Json encoding.

Reply:
However there may be a problem, because you use SType form patten matching, there as all data values should be described using RType. RType hierarchy is more precise, in particular it has distinct RType[Array[T]] and RType[Coll[T]]
where as SType don't.
Thus refactoring will be necessary to switch towards RType pattern matching.
I suggest to postpone it for the future PR